### PR TITLE
Add flag support to Nostrum for user public_flags

### PR DESCRIPTION
**Pull request by jos-b**
Hello Craig, hope you are safe.

I've implemented flags onto the user struct to allow people to see who is staff.

```elixir
iex> {:ok, %{public_flags: flags}} = Nostrum.Cache.UserCache.get(165023948638126080)
...
iex> flags
%Nostrum.Struct.Flags{
  bug_hunter_level_1: false,
  bug_hunter_level_2: false,
  early_supporter: true,
  hypesquad_balance: true,
  hypesquad_bravery: false,
  hypesquad_brilliance: false,
  hypesquad_events: false,
  partner: true,
  staff: false,
  system: false,
  team_user: false,
  verified_bot: false,
  verified_developer: true
}
iex> Nostrum.Struct.Flags.to_integer(flags)
131842
```

Thanks!

### CODE REVIEW

File: lib/nostrum/struct/user.ex [hunk 1]
```diff
@@ -63,6 +64,9 @@ defmodule Nostrum.Struct.User do
   @typedoc "The user's email"
   @type email :: String.t() | nil
 
+  @typedoc "The user's public flags"
+  @type public_flags :: Flags.t() | nil
```
Comment (skippi) [inline lib/nostrum/struct/user.ex:68]: Although the `nil` here matches discord's documentation, it doesn't quite match with the PR's code, which suggests that this value will never be `nil` (if I'm not mistaken). I'd remove the `nil` here to simplify pattern matching.

File: lib/nostrum/struct/flags.ex [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  iex> Nostrum.Struct.Flags.to_integer(my_flags)
+  131842
+  ```
+  """
+  @spec to_integer(t) :: integer()
+  def to_integer(flag_struct) do
+    booleans =
+      flag_struct
+      |> Map.to_list()
+      |> List.delete({:__struct__, Nostrum.Struct.Flags})
```
Comment (skippi) [inline lib/nostrum/struct/flags.ex:185]: I'd recommend using `Map.from_struct/1`.

File: lib/nostrum/struct/user.ex (hunk 1 above)
Comment (jos-b) [reply lib/nostrum/struct/user.ex]: Addressed in aff5d4ca76ec14983946c67521b8f03850f0a78c

File: lib/nostrum/struct/flags.ex (hunk 2 above)
Comment (jos-b) [reply lib/nostrum/struct/flags.ex]: Addressed in 2d4d439c6b77c8f2c6179a9b43a94184e93b9a28

File: lib/nostrum/struct/user.ex [hunk 3]
```diff
@@ -153,6 +158,7 @@ defmodule Nostrum.Struct.User do
       map
       |> Map.new(fn {k, v} -> {Util.maybe_to_atom(k), v} end)
       |> Map.update(:id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:public_flags, 0, &Flags.from_integer(&1))
```
Comment (Kraigie) [inline lib/nostrum/struct/user.ex:161]: If we set this to 0 here as a default then the spec will be wrong. I think we should leave the integer as it is, and if users want to act on the flags then they can use your function to do so.

Comment (jos-b) [reply lib/nostrum/struct/user.ex]: Craig! Thank you for the review. Addressed in 3cdff15fc4f6658d239630b8d669ffdc458fdeb8

**Pull request merged**